ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -884,14 +1029,25 @@ private GetResponse getMessagesForMissingKeys(ConnectedChannel connectedChannel,
           GetRequest.Replication_Client_Id_Prefix + dataNodeId.getHostname() + "[" + dataNodeId.getDatacenterName()
               + "]", MessageFormatFlags.All, partitionRequestInfoList,
           replicationConfig.replicationIncludeAll ? GetOption.Include_All : GetOption.None);
-      long startTime = SystemTime.getInstance().milliseconds();
+      long startTime = time.milliseconds();
       try {
         connectedChannel.send(getRequest);
         ChannelOutput channelOutput = connectedChannel.receive();
         getResponse = GetResponse.readFrom(new DataInputStream(channelOutput.getInputStream()), clusterMap);
-        long getRequestTime = SystemTime.getInstance().milliseconds() - startTime;
+        long getRequestTime = time.milliseconds() - startTime;
+        boolean remoteColoGetRequestForStandby = false;
+        if (leaderBasedReplicationAdmin != null && replicatingFromRemoteColo) {
+          // If leader-based replication is enabled, we should ideally send cross colo GET requests only between leaders.
+          // However, if standby replicas time out waiting for their keys to come from leader, we send cross colo GETs
+          // for them. Set 'remoteColoGetRequestForStandby' to true so that we track number of such cross colo GETs for standby replicas.
+          ReplicaId localReplica = replicasToReplicatePerNode.get(0).getLocalReplicaId();
+          ReplicaId remoteReplica = replicasToReplicatePerNode.get(0).getReplicaId();
+          if (!leaderBasedReplicationAdmin.isLeaderPair(localReplica, remoteReplica)) {
+            remoteColoGetRequestForStandby = true;
+          }
+        }
```

Comment:
I wonder why not put this logic in the code block starting from line 438 and pass in a boolean value to `fixMissingStoreKeys` and `getMessagesForMissingKeys`.  I see two benefits:
1. you don't have to if `leaderBasedReplicationAdmin = null` again;
2. you can skip some RemoteReplicaInfo associated with local leader replica;